File: java/solutions/src/test/java/io/github/ajoz/workshop/fp/tests/theory/Below.java
package io.github.ajoz.workshop.fp.tests.theory;

import org.junit.experimental.theories.ParametersSuppliedBy;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@ParametersSuppliedBy(BelowSupplier.class)
public @interface Below {
    int value();

    boolean inclusive() default false;

    int limit() default 1000;
}

File: kotlin/exercises/src/test/kotlin/io/github/ajoz/workshop/fp/kotlin/tests/theory/Below.kt
package io.github.ajoz.workshop.fp.kotlin.tests.theory

import org.junit.experimental.theories.ParametersSuppliedBy

@Retention(AnnotationRetention.RUNTIME)
@ParametersSuppliedBy(BelowSupplier::class)
annotation class Below(
        val value: Int,
        val inclusive: Boolean = false,
        val limit: Int = 1000
)
